src/timing-utilities.js

```diff
@@ -224,32 +224,39 @@
   var cubicBezierRe = new RegExp('cubic-bezier\\(' + numberString + ',' + numberString + ',' + numberString + ',' + numberString + '\\)');
   var stepRe = /steps\(\s*(\d+)\s*,\s*(start|middle|end)\s*\)/;
 
-  function toTimingFunction(easing) {
+  function normalizeEasing(easing) {
     if (!styleForCleaning) {
       styleForCleaning = document.createElement('div').style;
     }
     styleForCleaning.animationTimingFunction = '';
     styleForCleaning.animationTimingFunction = easing;
-    var validatedEasing = styleForCleaning.animationTimingFunction;
-
-    if (validatedEasing == '' && isInvalidTimingDeprecated()) {
+    var normalizedEasing = styleForCleaning.animationTimingFunction;
+    if (normalizedEasing == '' && isInvalidTimingDeprecated()) {
       throw new TypeError(easing + ' is not a valid value for easing');
     }
+    return normalizedEasing;
+  }
 
-    var cubicData = cubicBezierRe.exec(validatedEasing);
+  function parseEasingFunction(normalizedEasing) {
+    if (normalizedEasing == 'linear') {
+      return linear;
+    }
+    var cubicData = cubicBezierRe.exec(normalizedEasing);
     if (cubicData) {
       return cubic.apply(this, cubicData.slice(1).map(Number));
     }
-    var stepData = stepRe.exec(validatedEasing);
+    var stepData = stepRe.exec(normalizedEasing);
     if (stepData) {
       return step(Number(stepData[1]), {'start': Start, 'middle': Middle, 'end': End}[stepData[2]]);
     }
-    var preset = presets[validatedEasing];
+    var preset = presets[normalizedEasing];
     if (preset) {
       return preset;
     }
+    // Easing is invalid, this should never be reached.
```

Comment:
Nit: Elaborate this comment slightly (or add a comment to the start of the function) to explain that this is because parseEasingFunction has as a precondition that it be passed a valid, normalizedEasing string.